### PR TITLE
docs: Document UnknownExplanation.

### PR DESCRIPTION
**Pull request by mpreiner**
@4tXJ7f or @ajreynol can you please fill in the ??? blank?

### CODE REVIEW

File: src/api/cpp/cvc5_types.h [hunk 1]
```diff
@@ -28,15 +28,28 @@ namespace cvc5 {
...
   INCOMPLETE,
+  /** Time limit reached. */
   TIMEOUT,
+  /** Resource limit reached. */
   RESOURCEOUT,
+  /** Memory limit reached. */
   MEMOUT,
+  /** Solver was interrupted. */
   INTERRUPTED,
+  /** ??? */
```
Comment (ajreynol) [inline src/api/cpp/cvc5_types.h:46]: This corresponds to something from the optimization solver, I'm not sure what this means.

File: src/api/cpp/cvc5_types.h [hunk 2]
```diff
@@ -28,15 +28,28 @@ namespace cvc5 {
  */
 enum UnknownExplanation
 {
+  /**
+   * Full satisfiability check required (e.g., if only preprocessing was
+   * performed).
+   */
   REQUIRES_FULL_CHECK,
+  /** Incomplete theory. */
```
Comment (4tXJ7f) [inline src/api/cpp/cvc5_types.h:36]:
```suggestion
  /** Incomplete theory solver. */
```

is maybe a bit more accurate?

File: src/api/cpp/cvc5_types.h (hunk 1 above)
Comment (4tXJ7f) [reply src/api/cpp/cvc5_types.h:46]: I think that it can be removed like so #8518.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**